obspy/io/nordic/ellipse.py

```diff
@@ -0,0 +1,504 @@
+        :type dist: float
+        :param baz:   back-azimuth from observer (degrees)
+        :type baz: float
+        :param viewpoint: observer's position
+        :type viewpoint: 2-tuple of floats
+        :return: ellipse
+        :rtype: :class: `~obspy.io.nordic.ellipse.Ellipse`
+        """
+        x = viewpoint[0] + dist * np.sin(np.radians(baz))
+        y = viewpoint[1] + dist * np.cos(np.radians(baz))
+        return cls.from_uncerts(x_err, y_err, c_xy, (x, y))
+
+    def _almost_good_cov(cov):
+        """Checks if a covariance matrix is "almost good" (c_xy is greater
+        than c_xx and/or c_yy, but only by a little bit)
+        """
+        if cov[0][1] != cov[1][0]:
+            return False
+        ratio = cov[0][1]**2 / (cov[0][0] * cov[1][1])
+        if (ratio > 1) and (ratio < 1.1):
+            return True
+        return False
+
+    def _fix_cov(cov):
+        """Correct an "almost good" covariance matrix by making c_xx*c_yy
+        > c_xy^2
+        """
+        if not Ellipse._almost_good_cov(cov):
+            return None
+        ratio = cov[0][1]**2 / (cov[0][0] * cov[1][1])
+        cov[0][0] *= np.sqrt(ratio)*1.01
+        cov[1][1] *= np.sqrt(ratio)*1.01
+        return cov
+
+    def to_cov(self):
+        """Convert to covariance matrix notation
+
+        Sources:
+            https://stackoverflow.com/questions/41807958/
+                    convert-position-confidence-ellipse-to-covariance-matrix
+
+        :returns: covariance_matrix [[c_xx, c_xy], [c_xy, c_yy]],
+                 center_position (x,y)
+        :rtype: 2-tuple
+        """
+        sin_theta = np.sin(np.radians(self.theta))
+        cos_theta = np.cos(np.radians(self.theta))
+        # The following is BACKWARDs from math standard
+        # but consistent with geographic notation
+        c_yy = self.a**2 * cos_theta**2 + self.b**2 * sin_theta**2
+        c_xx = self.a**2 * sin_theta**2 + self.b**2 * cos_theta**2
+        c_xy = (self.a**2 - self.b**2) * sin_theta * cos_theta
+        return [[c_xx, c_xy], [c_xy, c_yy]], (self.x, self.y)
+
+    def to_uncerts(self):
+        """Convert to Nordic uncertainty values
+
+        Call as x_err, y_err, c_xy, center =myellipse.to_uncerts()
+
+        :parm self: ellipse
+        :type self: :class: `~obspy.io.nordic.ellipse.Ellipse`
+        :returns: x_error (m), y_error (m),
+                  x-y covariance (dist^2), center (x,y)
+        :rtype: 4-tuple
+        """
+        cov, center = self.to_cov()
+        assert cov[0][1] == cov[1][0]
+        x_err = np.sqrt(cov[0][0])
+        y_err = np.sqrt(cov[1][1])
+        c_xy = cov[0][1]
+        return x_err, y_err, c_xy, center
+
+    def __repr__(self):
+        """String describing the ellipse
+
+        >>> str(Ellipse(20,10))
+        '<a=20, b=10, theta=  0.0>'
+        >>>str(Ellipse(20,10,45,(3,4)))
+        '<a=20, b=10, theta= 45.0, center=(3,4)>'
+        """
+        s = f'<a={self.a:.3g}, b={self.b:.3g}'
```

Comment:
I'm not familiar with this syntax, and circleci thinks there is a syntax error here (which is resulting in appveyor and travis failing to import this file, and then the tests don't run).
In general the `__repr__` method should return a string formatted in the same way that the object is instantiated (see [the docs here](https://docs.python.org/3/reference/datamodel.html#object.__repr__), e.g. this might be better returning a string formatted like:
```python
str = "Ellipse(a={a}, b={b}, theta={theta}, center={center})".format(
    self.a, self.b, self.theta, self.center)
```

Reply:
Actually: this might work:
```python

    def __repr__(self):
        """String describing the ellipse

        >>> print(Ellipse(20, 10))
        Ellipse(a=20, b=10, theta=0.0, center=(0, 0))
        >>> print(Ellipse(20, 10, 45, (3, 4)))
        Ellipse(a=20, b=10, theta=45.0, center=(3,4))
        """
        s = ("Ellipse(a={a:.3g}, b={b:.3g}, theta={theta:.1f},"
             " center=({x:.3g}, {y:.3g}))".format(
                 a=self.a, b=self.b, theta=self.theta,
                 x=self.x, y=self.y))
        return s
```

Reply:
Regarding the binary files that appear in git status: it's not a mode change, but an internal CR/LF change:
```
obspy 508> git diff obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz
warning: CRLF will be replaced by LF in obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz.
The file will have its original line endings in your working directory.
diff --git a/obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz b/obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz
index d15b54c..66d4101 100644
Binary files a/obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz and b/obspy/signal/tests/data/KARC.LHZ.SAC.asc.gz differ
```

Reply:
As for the ```f'...'``` notation, it's new in Python 3.6 and circlei doesn't seem to like it (probably hurts back-compatibility).  I've gone back to the ```'...'.format(...)``` notation